src/Process.php

```diff
@@ -110,13 +110,24 @@ public function start(LoopInterface $loop, $interval = 0.1)
             stream_set_blocking($pipe, 0);
         }
 
-        $loop->addPeriodicTimer($interval, function (Timer $timer) {
+        $periodicMonitor = function (Timer $timer) {
             if (!$this->isRunning()) {
                 $this->close();
                 $timer->cancel();
                 $this->emit('exit', array($this->getExitCode(), $this->getTermSignal()));
             }
-        });
+        };
+
+        $timer = $loop->addPeriodicTimer($interval, $periodicMonitor);
+
+        $dataMonitor = function ($data) use ($timer, $periodicMonitor) {
+            if ('' === $data) {
+                $periodicMonitor($timer);
+            }
+        };
+
+        $this->stdout->on('data', $dataMonitor);
+        $this->stderr->on('data', $dataMonitor);
```

Comment:
> As far as I tested correctly, there is no end event that we could bound to

Indeed, I'd just filed ticket #17 :-)

I think it makes sense to tackle that first and then implement this feature on top of that.